releases/release-9.0.0.md

```diff
@@ -0,0 +1,355 @@
+
+    For more information, see [documentation](/statement-summary-tables.md#statements_summary-fields-description).
+
+* Optimize the `execution info` in the output of `EXPLAIN ANALYZE` [#56232](https://github.com/pingcap/tidb/issues/56232) @[yibin87](https://github.com/yibin87) tw@hfxsd<!--1697-->
+
+    [`EXPLAIN ANALYZE`](https://github.com/sql-statements/sql-statement-explain-analyze.md) executes SQL statements and records execution details in the `execution info` column. The same information is captured in the [slow query log](https://github.com/identify-slow-queries.md). These details are crucial for analyzing and understanding the time spent on SQL execution.
+
+    In v9.0.0, the `execution info` output is optimized for clearer representation of each metric. For example, `time` now refers to the wall-clock time for operator execution, `loops` indicates how many times the current operator is called by its parent operator, and `total_time` represents the cumulative duration of all concurrent executions. These optimizations help you better understand the SQL execution process and devise more targeted optimization strategies.
+
+    For more information, see [documentation](/sql-statements/sql-statement-explain-analyze.md).
+
+### Security
+
+### Data migration
+
+* Support query argument redaction in DM logs [#11489](https://github.com/pingcap/tiflow/issues/11489) @[db-will](https://github.com/db-will) **tw@Oreoxmt** <!--2030-->
+
+    Starting from v9.0.0, you can use the `redact-info-log` configuration item to enable the DM log redaction feature. When enabled, query arguments that contain sensitive data in DM logs are replaced with the `?` placeholder. To enable this feature, set `redact-info-log` to `true` in the DM-worker configuration file or pass `--redact-info-log=true` when starting DM. This feature only desensitizes query arguments, not the entire SQL statement, and requires a DM-worker restart to take effect.
+
+    For more information, see [documentation](/dm/dm-worker-configuration-file.md#redact-info-log-new-in-v900).
+
+* TiDB Lightning supports compatibility with `sql_require_primary_key=ON` in TiDB [#57479](https://github.com/pingcap/tidb/issues/57479) @[lance6716](https://github.com/lance6716) **tw@Oreoxmt** <!--2026-->
+
+    When you enable the system variable [`sql_require_primary_key`](/system-variables.md#sql_require_primary_key-new-in-v630) in TiDB, tables must have a primary key. To prevent table creation failures, TiDB Lightning adds a default primary key to its internal error-logging and conflict-detection tables (`conflict_error_v4`, `type_error_v2`, and `conflict_records_v2`). If you have automation scripts that use these internal tables, update them to accommodate the new schema, which includes the primary key.
```

Comment:
```suggestion
    When the system variable [`sql_require_primary_key`](/system-variables.md#sql_require_primary_key-new-in-v630) is enabled in TiDB, tables are required to have a primary key. To avoid table creation failures, TiDB Lightning adds a default primary key to its internal error-logging and conflict-detection tables (`conflict_error_v4`, `type_error_v2`, and `conflict_records_v2`). If you have automation scripts that depend on these internal tables, update them to accommodate the new schema, which now includes a primary key.
```